internal/controller/nonadminbackupstoragelocation_controller.go

```diff
@@ -377,6 +413,180 @@ func (r *NonAdminBackupStorageLocationReconciler) setFinalizerOnNaBSL(ctx contex
 	return false, nil
 }
 
+// ensureNonAdminRequest updates the NonAdminBackupStorageLocation object based on the
+// cluster admin's approval decision on the NonAdminBackupStorageLocationRequest object
+// and ensures Velero BackupStorageLocation and secret are deleted if the approval decision
+// is rejected
+func (r *NonAdminBackupStorageLocationReconciler) ensureNonAdminRequest(
+	ctx context.Context, logger logr.Logger, nabsl *nacv1alpha1.NonAdminBackupStorageLocation) (bool, error) {
+	veleroObjectsNACUUID := nabsl.Status.VeleroBackupStorageLocation.NACUUID
+
+	nabslRequest, err := function.GetNabslRequestByLabel(ctx, r.Client, r.OADPNamespace, veleroObjectsNACUUID)
+	if err != nil {
+		logger.Error(err, findSingleNABSLRequestError)
+		return false, err
+	} else if nabslRequest == nil {
+		err := errors.New("no NonAdminBackupStorageLocationRequest found")
+		logger.Error(err, findSingleNABSLRequestError)
+		return false, err
+	}
+
+	var terminalErr error
+	var reason, message string
+
+	adminApprovedCondition := metav1.ConditionFalse
+	expectedPhase := nacv1alpha1.NonAdminPhaseNew
+	updatedRejectedCondition := false
+	updatedApprovedCondition := false
+
+	if !reflect.DeepEqual(nabslRequest.Status.NonAdminBackupStorageLocationRequestStatusInfo.DeepCopy().RequestedSpec, nabsl.Spec.BackupStorageLocationSpec) {
+		message = "NaBSL Spec update not allowed. Changes will not be applied. Delete NaBSL and create new one with updated spec"
+		updatedRejectedCondition = meta.SetStatusCondition(&nabsl.Status.Conditions, metav1.Condition{
+			Type:    string(nacv1alpha1.NonAdminBSLConditionSpecUpdateApproved),
+			Status:  metav1.ConditionFalse,
+			Reason:  "BslSpecUpdateRejected",
+			Message: message,
+		})
+		adminApprovedCondition = metav1.ConditionTrue
```

Comment:
since this error case, shouldn't it be false and BackingOff?

Reply:
No.

In this scenario we have two conditions:

```go
	NonAdminBSLConditionApproved           NonAdminBSLCondition = "ClusterAdminApproved"
	NonAdminBSLConditionSpecUpdateApproved NonAdminBSLCondition = "SpecUpdateApproved"
```

This flag however is not used for updating any of the above conditions from (line number 480), it is set to true to not delete previously created BSL nor Secret on line number 490. This was added here and I can introduce new flag e.g. `deleteNacBslSecret`, but thought to use it to not have too many vars in this logic

Reply:
added new flag `preserveVeleroBslSecret` in the go code, to make it easier to read the workflow. Thanks.

Reply:
https://github.com/migtools/oadp-non-admin/pull/216/commits/3cf8e01c6a3d93bc6c0376fc390d136b6d2132af